encointer v0.7.0

clangenb opened the pull request:
Notable changes:
* [Dart] does no longer handle community identifier as a simple string
* [Dart] the location's latitude and longitude are no longer hex-strings, but strings of the format of `35.2413541`. Ultimately, we need to decide how to handle discrepancies that might occur in the current implementation of the string to fixed-point conversions: https://github.com/encointer/encointer-js/issues/35

* Closes #253 
* Closes #245 
* Closes #224 
* Closes #175 

Tested:
* Community Currency Transfer
* Registering participants with ProofOfAttendance
* Tested with successful meetup

clangenb commented on lib/js_service_encointer/.browserslistrc:
```diff
@@ -1,5 +1,5 @@
 > 0.1%
-android > 0
+android > 3
```
The app itself does not support < 4

clangenb commented on .github/workflows/js_ci.yml:
```diff
@@ -49,28 +49,28 @@ jobs:
-          cd lib/js_service_encointer
-          yarn install
-          yarn test:${{ matrix.network }}
+#  e2e-tests:
```
Gesell is no longer supported until we upgrade it.

clangenb commented on lib/utils/format.dart:
```diff
@@ -313,12 +308,7 @@ class Fmt {
-  }
-
-  /// Formats fixed point number with the amount of fractional digits given by [fixedPointFraction].
-  static String degreeFromHex(String degree, {int fixedPointFraction = 64, int fractionDisplay = 3}) {
```
We no longer need this as the location does already contain the float as string like `35.234234`

brenzi commented:
why do you re-introduce PIN dialog? We could ask for the PIN at the first time it is needed an cache it then

clangenb commented:
There were some issues when asking for the pin when getting the ProofOfAttendance, but I fixed it now, and reverted the change. PIN is only asked when necessary and is cached.

brenzi reviewed (approved):
tested with local setup. only tested app startup and basics

anizeani commented:
This pr breaks the pin check.. entering a wrong pin ends up in skipping the passwordDialog (see changes in assets/index.dart)
This pr does not show anymore the error message when wrong pin is entered (see components/passwordInputDialog.dart) because of the way Navigator.pop() is done
